x-pack/plugins/security_solution/public/common/components/event_details/osquery_tab.tsx

```diff
@@ -98,58 +92,57 @@ export const useOsqueryTab = ({
     return;
   }
 
-  const { OsqueryResults } = osquery;
   const expandedEventFieldsObject = expandDottedObject(
     rawEventData.fields
   ) as ExpandedEventFieldsObject;
 
-  const parameters = expandedEventFieldsObject.kibana?.alert?.rule?.parameters;
-  const responseActions = parameters?.[0].response_actions;
+  const responseActions =
+    expandedEventFieldsObject?.kibana?.alert?.rule?.parameters?.[0].response_actions;
 
-  const osqueryActionsLength = responseActions?.filter(
-    (action: { action_type_id: string }) => action.action_type_id === RESPONSE_ACTION_TYPES.OSQUERY
-  )?.length;
-
-  if (!osqueryActionsLength) {
+  if (!responseActions?.length) {
     return;
   }
-  const ruleName = expandedEventFieldsObject.kibana?.alert?.rule?.name;
-  const agentIds = expandedEventFieldsObject.agent?.id;
+
+  const { OsqueryResults, fetchAllLiveQueries } = osquery;
 
   const alertId = rawEventData._id;
 
+  const { data: actionsData } = fetchAllLiveQueries({
```

Comment:
what if there are more than 100 actions for this `alert_id`?

Reply:
I'll take it out, set to 0.

Reply:
Ups, 0 breaks it.

Reply:
I think we may leave it as 100 for now, or any other number that you prefer. And add infinite loop in the next PR.